api/konnect/v1alpha1/konnect_cloudgateway_dataplane_configuration_types.go

```diff
@@ -0,0 +1,257 @@
+	NetworkRefKonnectID = "konnectID"
+)
+
+// NetworkRef is the schema for the NetworkRef type.
+// It is used to reference a Network entity.
+//
+// +kubebuilder:object:generate=true
+// +kubebuilder:validation:XValidation:rule="(has(self.type) && self.type == 'konnectID') ? has(self.konnectID) : true", message="when type is konnectID, konnectID must be set"
+// +apireference:kgo:include
+type NetworkRef struct {
+	// Type indicates the type of the control plane being referenced.
+	//
+	// +kubebuilder:validation:Enum=konnectID
+	Type string `json:"type,omitempty"`
+
+	// KonnectID is the schema for the KonnectID type.
+	// This field is required when the Type is konnectID.
+	// +optional
+	KonnectID *string `json:"konnectID,omitempty"`
+}
+
+// KonnectConfigurationDataPlaneGroup is the schema for the KonnectConfiguration type.
+type KonnectConfigurationDataPlaneGroup struct {
+	// Name of cloud provider.
+	//
+	// +kubebuilder:validation:Required
+	Provider sdkkonnectcomp.ProviderName `json:"provider"`
+
+	// Region for cloud provider region.
+	//
+	// +kubebuilder:validation:Required
+	Region string `json:"region"`
+
+	// NetworkRef is the schema for the NetworkRef type.
+	//
+	// +kubebuilder:validation:Required
+	NetworkRef NetworkRef `json:"networkRef"`
+
+	// Autoscale configuration for the data-plane group.
+	//
+	// +kubebuilder:validation:Required
+	Autoscale ConfigurationDataPlaneGroupAutoscale `json:"autoscale"`
+
+	// Array of environment variables to set for a data-plane group.
+	//
+	// +kubebuilder:validation:Optional
+	Environment []ConfigurationDataPlaneGroupEnvironmentField `json:"environment,omitempty"`
+}
+
+// ConfigurationDataPlaneGroupEnvironmentField specifies an environment variable field for the data-plane group.
+type ConfigurationDataPlaneGroupEnvironmentField struct {
+	// Name of the environment variable field to set for the data-plane group. Must be prefixed by KONG_.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=63
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern="^KONG_."
+	Name string `json:"name"`
+	// Value assigned to the environment variable field for the data-plane group.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=63
+	// +kubebuilder:validation:Required
+	Value string `json:"value"`
+}
+
+// ConfigurationDataPlaneGroupAutoscale specifies the autoscale configuration for the data-plane group.
+//
+// +kubebuilder:validation:XValidation:rule="self.type == 'ConfigurationDataPlaneGroupAutoscaleStatic' ? has(self.static) : true",message="static is required when type is ConfigurationDataPlaneGroupAutoscaleStatic"
+// +kubebuilder:validation:XValidation:rule="self.type == 'ConfigurationDataPlaneGroupAutoscaleAutopilot' ? has(self.autopilot) : true",message="autopilot is required when type is ConfigurationDataPlaneGroupAutoscaleAutopilot"
+type ConfigurationDataPlaneGroupAutoscale struct {
```

Comment:
We may want to cross-validate that when the type is autopilot, static is not set, and viceversa.